ext/standard/ftp_fopen_wrapper.c

```diff
@@ -1070,41 +1070,53 @@ static int php_stream_ftp_mkdir(php_stream_wrapper *wrapper, const char *url, in
 
         /* find a top level directory we need to create */
         while ((p = strrchr(buf, '/'))) {
-            *p = '\0';
-			php_stream_printf(stream, "CWD %s\r\n", buf);
-			result = GET_FTP_RESULT(stream);
-			if (result >= 200 && result <= 299) {
-				*p = '/';
+			*p = '\0';
+			if (strlen(buf) == 0) {
+				php_stream_printf(stream, "CWD /\r\n");
+				result = GET_FTP_RESULT(stream);
+				if (result >= 200 && result <= 299) {
+					*p = '/';
+				}
 				break;
+			} else {
+				php_stream_printf(stream, "CWD %s\r\n", buf);
+				result = GET_FTP_RESULT(stream);
+				if (result >= 200 && result <= 299) {
+					*p = '/';
+					break;
+				}
```

Comment:
We could simplify this a bit if it's okay to issue `CWD /foo/bar/` instead of `CWD /foo/bar`, as `/` would no longer be a special case then. Do you know if the FTP spec allows that?

Reply:
I guess this wouldn't actually make things simpler because this is modifying the string in place and would end up overwriting a byte of the directory rather than a '/`.

Reply:
In that case, I'd only suggest to combined these two cases:

```
php_stream_printf(stream, "CWD %s\r\n", strlen(buf) ? buf : "/");
result = GET_FTP_RESULT(stream);
if (result >= 200 && result <= 299) {
    *p = '/';
    break;
}
```

Reply:
Awesome!
What do you think of the new changes?